File: AQS.Api/AQS.Reading.Domain/Models/Reading.cs
using System;

namespace AQS.Api.Reading.Domain.Models
{
    public class Reading : BaseEntity
    {
        public Reading(Guid deviceId, ReadingType readingType, string value)
        {
            DeviceId = deviceId;
            ReadingType = readingType;
            Value = value ?? throw new ArgumentNullException(nameof(value));
        }

        public Guid DeviceId { get; set; }
        public ReadingType ReadingType { get; set; }
        public string Value { get; set; }
    }
}

File: AQS.Arduino/Reading.cpp
/*
  Reading.cpp - Class representing a reading
  sent to the AQS api.
*/

#include "Arduino.h"
#include "Reading.h"

Reading::Reading(String deviceId, int type, String value)
{
  _deviceId = deviceId;
  _type = type;
  _value = value;
}
